With this project we will explore ebay's used cars sale data and will find any pattern in sell. If some kind of cars sold more often or quicker.

In [1]:
import numpy as np
import pandas as df
autos= df.read_csv("autos.csv", encoding ="Latin-1")
autos.head()

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  yearOfRegistration  \
0  privat   Angebot  $5,000  control         bus                2004   
1  privat   Angebot  $8,500  control   limousine                1997   
2  privat   Angebot  $8,990     test   limousine                2009   
3  privat   Angebot  $4,350  control  kleinwagen                2007   
4  privat   Angebot  $1,350     test       kombi                2003   

     gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0    manuell      158  andere  150,000km                    3      lpg   
1  automatik      286     7er  150,000km                    6   benzin   
2    manuell      102    golf   70,000km                    7   benzin   
3  automatik       71  fortwo   70,000km                    6   benzin   
4    manuell        0   focus  150,000km                    7   benzin   

        brand notRepairedDamage          dateCreated  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50

In [2]:
autos

dateCrawled                                               name  \
0      2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1      2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2      2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3      2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4      2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
5      2016-03-21 13:47:45  Chrysler_Grand_Voyager_2.8_CRD_Aut.Limited_Sto...   
6      2016-03-20 17:55:21  VW_Golf_III_GT_Special_Electronic_Green_Metall...   
7      2016-03-16 18:55:19                               Golf_IV_1.9_TDI_90PS   
8      2016-03-22 16:51:34                                         Seat_Arosa   
9      2016-03-16 13:47:02          Renault_Megane_Scenic_1.6e_RT_Klimaanlage   
10     2016-03-15 01:41:36                       VW_Golf_Tuning_in_siber/grau   
11     2016-03-16 18:45:34                         Mercedes_A140_Motorschaden   
12     2016-03-31 19:48:22  Smart_smart_fortwo_coupe_softouch_pure_MHD_Pan...   
13     2016-03-23 10:48:32                                 Audi_A3_1.6_tuning   
14     2016-03-23 11:50:46  Renault_Clio_3__Dynamique_1.2__16_V;_viele_Ver...   
15     2016-04-01 12:06:20        Corvette_C3_Coupe_T_Top_Crossfire_Injection   
16     2016-03-16 14:59:02                                Opel_Vectra_B_Kombi   
17     2016-03-29 11:46:22                          Volkswagen_Scirocco_2_G60   
18     2016-03-26 19:57:44               Verkaufen_mein_bmw_e36_320_i_touring   
19     2016-03-17 13:36:21        mazda_tribute_2.0_mit_gas_und_tuev_neu_2018   
20     2016-03-05 19:57:31  Audi_A4_Avant_1.9_TDI_*6_Gang*AHK*Klimatronik*...   
21     2016-03-06 19:07:10                      Porsche_911_Carrera_4S_Cabrio   
22     2016-03-28 20:50:54                               MINI_Cooper_S_Cabrio   
23     2016-03-10 19:55:34   Peugeot_Boxer_2_2_HDi_120_Ps_9_Sitzer_inkl_Klima   
24     2016-04-03 11:57:02                         BMW_535i_xDrive_Sport_Aut.   
25     2016-03-21 21:56:18  Ford_escort_kombi_an_bastler_mit_ghia_ausstattung   
26     2016-04-03 22:46:28                                Volkswagen_Polo_Fox   
27     2016-03-27 18:45:01             Hat_einer_Ahnung_mit_Ford_Galaxy_HILFE   
28     2016-03-19 21:56:19                                      MINI_Cooper_D   
29     2016-04-02 12:45:44  Mercedes_Benz_E_320_T_CDI_Avantgarde_DPF7_Sitz...   
...                    ...                                                ...   
49970  2016-03-21 22:47:37  c4_Grand_Picasso_mit_Automatik_Leder_Navi_Temp...   
49971  2016-03-29 14:54:12                                         W.Lupo_1.0   
49972  2016-03-26 22:25:23          Mercedes_Benz_Vito_115_CDI_Extralang_Aut.   
49973  2016-03-27 05:32:39                   Mercedes_Benz_SLK_200_Kompressor   
49974  2016-03-20 10:52:31  Golf_1_Cabrio_Tuev_Neu_viele_Extras_alles_eing...   
49975  2016-03-27 20:51:39          Honda_Jazz_1.3_DSi_i_VTEC_IMA_CVT_Comfort   
49976  2016-03-19 18:56:05  Audi_80_Avant_2.6_E__Vollausstattung!!_Einziga...   
49977  2016-03-31 18:37:18                        Mercedes_Benz_C200_Cdi_W203   
49978  2016-04-04 10:37:14                        Mercedes_Benz_E_200_Classic   
49979  2016-03-20 18:38:40                      Volkswagen_Polo_1.6_TDI_Style   
49980  2016-03-12 10:55:54                            Ford_Escort_Turnier_16V   
49981  2016-03-15 09:38:21             Opel_Astra_Kombi_mit_Anhaengerkupplung   
49982  2016-03-29 18:51:08            Skoda_Fabia_4_Tuerer_Bj:2004__85.000Tkm   
49983  2016-03-06 12:43:04                                      Ford_focus_99   
49984  2016-03-31 22:48:48  Student_sucht_ein__Anfaengerauto___ab_2000_BJ_...   
49985  2016-04-02 16:38:23                          Verkaufe_meinen_vw_vento!   
49986  2016-04-04 20:46:02  Chrysler_300C_3.0_CRD_DPF_Automatik_Voll_Ausst...   
49987  2016-03-22 20:47:27  Audi_A3_Limousine_

In [3]:
autos.info()
autos.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
dateCrawled            50000 non-null object
name                   50000 non-null object
seller                 50000 non-null object
offerType              50000 non-null object
price                  50000 non-null object
abtest                 50000 non-null object
vehicleType            44905 non-null object
yearOfRegistration     50000 non-null int64
gearbox                47320 non-null object
powerPS                50000 non-null int64
model                  47242 non-null object
odometer               50000 non-null object
monthOfRegistration    50000 non-null int64
fuelType               45518 non-null object
brand                  50000 non-null object
notRepairedDamage      40171 non-null object
dateCreated            50000 non-null object
nrOfPictures           50000 non-null int64
postalCode             50000 non-null int64
lastSeen               50000 non-null obj

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  yearOfRegistration  \
0  privat   Angebot  $5,000  control         bus                2004   
1  privat   Angebot  $8,500  control   limousine                1997   
2  privat   Angebot  $8,990     test   limousine                2009   
3  privat   Angebot  $4,350  control  kleinwagen                2007   
4  privat   Angebot  $1,350     test       kombi                2003   

     gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0    manuell      158  andere  150,000km                    3      lpg   
1  automatik      286     7er  150,000km                    6   benzin   
2    manuell      102    golf   70,000km                    7   benzin   
3  automatik       71  fortwo   70,000km                    6   benzin   
4    manuell        0   focus  150,000km                    7   benzin   

        brand notRepairedDamage          dateCreated  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50

In [4]:
autos.columns


Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [5]:
autos.columns=['date_crawled', 'name', 'seller', 'offer_type', 'price', 'abtest',
       'vehicle_type', 'registration_year', 'gearbox', 'power_ps', 'model',
       'odometer', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damage', 'ad_created', 'number_of_pictures', 'postal_code',
       'last_seen']

In [6]:
autos.head()

date_crawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offer_type   price   abtest vehicle_type  registration_year  \
0  privat    Angebot  $5,000  control          bus               2004   
1  privat    Angebot  $8,500  control    limousine               1997   
2  privat    Angebot  $8,990     test    limousine               2009   
3  privat    Angebot  $4,350  control   kleinwagen               2007   
4  privat    Angebot  $1,350     test        kombi               2003   

     gearbox  power_ps   model   odometer  registration_month fuel_type  \
0    manuell       158  andere  150,000km                   3       lpg   
1  automatik       286     7er  150,000km                   6    benzin   
2    manuell       102    golf   70,000km                   7    benzin   
3  automatik        71  fortwo   70,000km                   6    benzin   
4    manuell         0   focus  150,000km                   7    benzin   

        brand unrepaired_damage           ad_created  number_of_pictures  \
0     peugeot              nein  2016-03-26 00:00:00                   0   
1         bmw              nein  2016-04-04 00:00:00                   0   
2  volkswagen              nein  2016-03-26 00:00:00                   0   
3       smart              nein  2016-03-12 00:00:00                   0   
4        ford              nein  2016-04-01 00:00:00                   0   

   postal_code            last_seen  
0        79588  2016-04-06 06:45:54  
1        71034  2016-04-06 14:45:08  
2        35394  2016-04-06 20:15:37  
3        33729  2016-03-15 03:16:28  
4        39218  2016-04-01 14:38:50

Changed column names to snakecase, assigned back to dataframe.

In [7]:
autos.describe()

registration_year      power_ps  registration_month  \
count       50000.000000  50000.000000        50000.000000   
mean         2005.073280    116.355920            5.723360   
std           105.712813    209.216627            3.711984   
min          1000.000000      0.000000            0.000000   
25%          1999.000000     70.000000            3.000000   
50%          2003.000000    105.000000            6.000000   
75%          2008.000000    150.000000            9.000000   
max          9999.000000  17700.000000           12.000000   

       number_of_pictures   postal_code  
count             50000.0  50000.000000  
mean                  0.0  50813.627300  
std                   0.0  25779.747957  
min                   0.0   1067.000000  
25%                   0.0  30451.000000  
50%                   0.0  49577.000000  
75%                   0.0  71540.000000  
max                   0.0  99998.000000

* Number of picture column has 0 as the value, can drop that column.
* Registration year needs more investigation as minimum value is 1000 and maximum value is 9999

In [8]:
autos.rename(columns = {"odometer" : "odometer_km"}, inplace = True)

In [9]:
autos.head()

date_crawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offer_type   price   abtest vehicle_type  registration_year  \
0  privat    Angebot  $5,000  control          bus               2004   
1  privat    Angebot  $8,500  control    limousine               1997   
2  privat    Angebot  $8,990     test    limousine               2009   
3  privat    Angebot  $4,350  control   kleinwagen               2007   
4  privat    Angebot  $1,350     test        kombi               2003   

     gearbox  power_ps   model odometer_km  registration_month fuel_type  \
0    manuell       158  andere   150,000km                   3       lpg   
1  automatik       286     7er   150,000km                   6    benzin   
2    manuell       102    golf    70,000km                   7    benzin   
3  automatik        71  fortwo    70,000km                   6    benzin   
4    manuell         0   focus   150,000km                   7    benzin   

        brand unrepaired_damage           ad_created  number_of_pictures  \
0     peugeot              nein  2016-03-26 00:00:00                   0   
1         bmw              nein  2016-04-04 00:00:00                   0   
2  volkswagen              nein  2016-03-26 00:00:00                   0   
3       smart              nein  2016-03-12 00:00:00                   0   
4        ford              nein  2016-04-01 00:00:00                   0   

   postal_code            last_seen  
0        79588  2016-04-06 06:45:54  
1        71034  2016-04-06 14:45:08  
2        35394  2016-04-06 20:15:37  
3        33729  2016-03-15 03:16:28  
4        39218  2016-04-01 14:38:50

In [10]:
autos["price"] = autos["price"].str.replace(",", "").str.replace("$", "").astype(int)
autos["odometer_km"] = autos ["odometer_km"].str.replace(",", "").str.replace("km", "").astype(int)


In [11]:
autos.head(3)

date_crawled                                        name  seller  \
0  2016-03-26 17:47:46            Peugeot_807_160_NAVTECH_ON_BOARD  privat   
1  2016-04-04 13:38:56  BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik  privat   
2  2016-03-26 18:57:24                  Volkswagen_Golf_1.6_United  privat   

  offer_type  price   abtest vehicle_type  registration_year    gearbox  \
0    Angebot   5000  control          bus               2004    manuell   
1    Angebot   8500  control    limousine               1997  automatik   
2    Angebot   8990     test    limousine               2009    manuell   

   power_ps   model  odometer_km  registration_month fuel_type       brand  \
0       158  andere       150000                   3       lpg     peugeot   
1       286     7er       150000                   6    benzin         bmw   
2       102    golf        70000                   7    benzin  volkswagen   

  unrepaired_damage           ad_created  number_of_pictures  postal_code  \
0              nein  2016-03-26 00:00:00                   0        79588   
1              nein  2016-04-04 00:00:00                   0        71034   
2              nein  2016-03-26 00:00:00                   0        35394   

             last_seen  
0  2016-04-06 06:45:54  
1  2016-04-06 14:45:08  
2  2016-04-06 20:15:37

In [12]:
autos["price"].describe()

count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price, dtype: float64

In [13]:
autos["odometer_km"].unique().shape

(13,)

In [14]:
autos["odometer_km"].describe()

count     50000.000000
mean     125732.700000
std       40042.211706
min        5000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64

In [15]:
autos["odometer_km"].value_counts(ascending=False).head()

150000    32424
125000     5170
100000     2169
90000      1757
80000      1436
Name: odometer_km, dtype: int64

I do not see any outliers in odometer_km.

In [16]:
autos["price"] = autos ["price"].astype(float)

In [17]:
autos["price"].unique().shape

(2357,)

In [18]:
autos["price"].describe()

count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price, dtype: float64

In [19]:
autos["price"].value_counts(ascending=False).head()

0.0       1421
500.0      781
1500.0     734
2500.0     643
1200.0     639
Name: price, dtype: int64

In [20]:
autos = autos[autos["price"] != 0]

In [21]:
autos["price"].value_counts(ascending=False).head()

500.0     781
1500.0    734
2500.0    643
1200.0    639
1000.0    639
Name: price, dtype: int64

In [22]:
print(autos['date_crawled'].str[:10])

0        2016-03-26
1        2016-04-04
2        2016-03-26
3        2016-03-12
4        2016-04-01
5        2016-03-21
6        2016-03-20
7        2016-03-16
8        2016-03-22
9        2016-03-16
10       2016-03-15
11       2016-03-16
12       2016-03-31
13       2016-03-23
14       2016-03-23
15       2016-04-01
16       2016-03-16
17       2016-03-29
18       2016-03-26
19       2016-03-17
20       2016-03-05
21       2016-03-06
22       2016-03-28
23       2016-03-10
24       2016-04-03
25       2016-03-21
26       2016-04-03
28       2016-03-19
29       2016-04-02
30       2016-03-14
            ...    
49968    2016-04-01
49969    2016-03-17
49970    2016-03-21
49971    2016-03-29
49972    2016-03-26
49973    2016-03-27
49975    2016-03-27
49976    2016-03-19
49977    2016-03-31
49978    2016-04-04
49979    2016-03-20
49980    2016-03-12
49981    2016-03-15
49982    2016-03-29
49983    2016-03-06
49985    2016-04-02
49986    2016-04-04
49987    2016-03-22
49988    2016-03-28


In [23]:
print(autos['date_crawled'].value_counts(normalize = True, dropna=False))

2016-04-02 11:37:04    0.000062
2016-03-09 11:54:38    0.000062
2016-03-23 18:39:34    0.000062
2016-03-23 19:38:20    0.000062
2016-03-19 17:36:18    0.000062
2016-03-08 10:40:35    0.000062
2016-03-16 21:50:53    0.000062
2016-03-05 16:57:05    0.000062
2016-03-21 20:37:19    0.000062
2016-03-12 16:06:22    0.000062
2016-03-25 19:57:10    0.000062
2016-03-14 20:50:02    0.000062
2016-04-04 16:40:33    0.000062
2016-03-30 19:48:02    0.000062
2016-03-29 23:42:13    0.000062
2016-03-11 22:38:16    0.000062
2016-04-02 15:49:30    0.000062
2016-03-22 09:51:06    0.000062
2016-04-03 08:36:19    0.000041
2016-03-17 00:38:59    0.000041
2016-03-12 11:57:06    0.000041
2016-03-26 16:06:23    0.000041
2016-03-27 21:56:00    0.000041
2016-03-15 23:06:21    0.000041
2016-03-10 19:51:53    0.000041
2016-03-25 12:53:03    0.000041
2016-03-27 10:53:56    0.000041
2016-03-08 03:56:11    0.000041
2016-03-24 18:36:23    0.000041
2016-04-02 14:54:39    0.000041
                         ...   
2016-03-

In [24]:
print(autos['registration_year'].describe())

count    48579.000000
mean      2004.753000
std         88.631663
min       1000.000000
25%       1999.000000
50%       2004.000000
75%       2008.000000
max       9999.000000
Name: registration_year, dtype: float64


* As per I mentioned in earler observation, registration year has some outliers - as years can not be 1000 or 9999

In [25]:
autos["registration_year"].value_counts(ascending=True)

1952       1
1931       1
1000       1
1948       1
9000       1
1927       1
8888       1
4100       1
6200       1
2800       1
1929       1
1938       1
1943       1
5911       1
4800       1
1111       1
1953       1
4500       1
1939       1
1001       1
2019       2
1955       2
1957       2
1954       2
1934       2
1941       2
1951       2
1800       2
9999       3
1950       3
        ... 
1989     174
1991     339
1990     347
1992     371
2015     392
1993     425
2018     471
1994     629
2014     664
2013     803
2016    1220
1995    1228
2012    1310
1996    1373
2017    1393
2010    1589
2011    1623
1997    1951
2009    2086
2008    2215
2007    2277
1998    2363
2002    2486
2001    2637
2006    2671
2003    2699
2004    2703
1999    2899
2005    2936
2000    3156
Name: registration_year, Length: 95, dtype: int64

Registration year of the car can not be in future, hence will drop values > 2019
Same way, registration year can not be earlier than 1900 as car registration system started in 1901.

In [26]:
autos = autos[autos["registration_year"] >= 1900  ]

In [27]:
autos = autos[autos["registration_year"] <= 2019  ]

In [28]:
print(autos['registration_year'].value_counts(normalize = True))

2000    0.064993
2005    0.060463
1999    0.059701
2004    0.055664
2003    0.055582
2006    0.055005
2001    0.054305
2002    0.051195
1998    0.048662
2007    0.046891
2008    0.045615
2009    0.042958
1997    0.040178
2011    0.033423
2010    0.032723
2017    0.028687
1996    0.028275
2012    0.026977
1995    0.025289
2016    0.025124
2013    0.016537
2014    0.013674
1994    0.012953
2018    0.009700
1993    0.008752
2015    0.008073
1992    0.007640
1990    0.007146
1991    0.006981
1989    0.003583
          ...   
1976    0.000432
1969    0.000391
1975    0.000371
1965    0.000350
1964    0.000247
1963    0.000165
1959    0.000124
1961    0.000124
1910    0.000103
1937    0.000082
1962    0.000082
1958    0.000082
1956    0.000082
1950    0.000062
1955    0.000041
2019    0.000041
1934    0.000041
1957    0.000041
1951    0.000041
1941    0.000041
1954    0.000041
1953    0.000021
1948    0.000021
1927    0.000021
1943    0.000021
1929    0.000021
1939    0.000021
1938    0.0000

* Higest cars sold were registered in year 2000, followed by 2005.

In [29]:
autos["brand"].unique().shape

(40,)

In [30]:
autos["brand"].describe

<bound method Series.describe of 0               peugeot
1                   bmw
2            volkswagen
3                 smart
4                  ford
5              chrysler
6            volkswagen
7            volkswagen
8                  seat
9               renault
10           volkswagen
11        mercedes_benz
12                smart
13                 audi
14              renault
15       sonstige_autos
16                 opel
17           volkswagen
18                  bmw
19                mazda
20                 audi
21              porsche
22                 mini
23              peugeot
24                  bmw
25                 ford
26           volkswagen
28                 mini
29        mercedes_benz
30              peugeot
              ...      
49968     mercedes_benz
49969            nissan
49970           citroen
49971        volkswagen
49972     mercedes_benz
49973     mercedes_benz
49975             honda
49976              audi
49977     mercedes_benz
49978  

In [31]:
print(autos['brand'].value_counts(normalize = True))

volkswagen        0.212813
opel              0.108631
bmw               0.108631
mercedes_benz     0.095780
audi              0.085834
ford              0.069688
renault           0.047880
peugeot           0.029449
fiat              0.026010
seat              0.018925
skoda             0.016063
nissan            0.015260
mazda             0.015219
smart             0.014292
citroen           0.014127
toyota            0.012583
hyundai           0.009947
sonstige_autos    0.009700
volvo             0.009061
mini              0.008608
mitsubishi        0.008176
honda             0.007990
kia               0.007105
alfa_romeo        0.006611
porsche           0.005910
suzuki            0.005890
chevrolet         0.005663
chrysler          0.003480
dacia             0.002657
daihatsu          0.002512
jeep              0.002204
subaru            0.002101
land_rover        0.002039
saab              0.001627
daewoo            0.001565
jaguar            0.001524
trabant           0.001380
r

* Brand volkswagen is sold the most on ebay, follwed by bmw and opel.

In [32]:
brands_top_twenty = autos["brand"].value_counts().head(20)
mean_val = {}
brands = brands_top_twenty.index

for br in brands:
    avg = autos.loc[autos["brand"] == br, "price"].mean()
    mean_val[br] = round(avg, 2)
print(mean_val)    



{'seat': 4320.17, 'volvo': 32914.29, 'bmw': 8493.86, 'renault': 2431.2, 'audi': 9212.93, 'skoda': 6353.54, 'mazda': 4059.06, 'fiat': 12566.54, 'opel': 5281.32, 'nissan': 4669.39, 'peugeot': 3065.61, 'toyota': 5148.0, 'hyundai': 5371.79, 'citroen': 43578.91, 'ford': 7305.14, 'smart': 3518.1, 'mini': 10541.57, 'mercedes_benz': 30025.54, 'volkswagen': 6600.38, 'sonstige_autos': 44381.44}


* Even though volkswagen is the top sold cars in ebay, citroen has the highest selling price.

In [46]:
brands_top_twenty = autos["brand"].value_counts().head(20)
mean_val = {}
mean_mil = {}
brands = brands_top_twenty.index

for br in brands:
    avg = autos.loc[autos["brand"] == br, "price"].mean()
    avg_mil = autos.loc[autos["brand"]== br, "odometer_km" ].mean()
    mean_val[br] = round(avg, 2)
    mean_mil[br] = avg_mil


In [51]:
mean_val_series = df.Series(mean_val) 
mean_mil_series = df.Series(mean_mil)

df_mean = df.DataFrame(mean_val_series,columns=['mean_values'])
df_mean

mean_values
audi                9212.93
bmw                 8493.86
citroen            43578.91
fiat               12566.54
ford                7305.14
hyundai             5371.79
mazda               4059.06
mercedes_benz      30025.54
mini               10541.57
nissan              4669.39
opel                5281.32
peugeot             3065.61
renault             2431.20
seat                4320.17
skoda               6353.54
smart               3518.10
sonstige_autos     44381.44
toyota              5148.00
volkswagen          6600.38
volvo              32914.29

In [52]:
#mean_val_series = df.Series(mean_val) 
#mean_mil_series = df.Series(mean_mil)

df_mean['mean_odo'] = df.DataFrame(mean_mil_series)
df_mean

mean_values       mean_odo
audi                9212.93  129492.562380
bmw                 8493.86  132686.255924
citroen            43578.91  120043.731778
fiat               12566.54  117593.032462
ford                7305.14  124327.718676
hyundai             5371.79  106718.426501
mazda               4059.06  124871.447903
mercedes_benz      30025.54  130852.504838
mini               10541.57   88899.521531
nissan              4669.39  118711.201080
opel                5281.32  129455.924171
peugeot             3065.61  127356.643357
renault             2431.20  128062.365591
seat                4320.17  121768.226333
skoda               6353.54  110993.589744
smart               3518.10  100511.527378
sonstige_autos     44381.44   90467.091295
toyota              5148.00  116219.312602
volkswagen          6600.38  128961.679892
volvo              32914.29  138363.636364

In [54]:
print(df_mean)

                mean_values       mean_odo
audi                9212.93  129492.562380
bmw                 8493.86  132686.255924
citroen            43578.91  120043.731778
fiat               12566.54  117593.032462
ford                7305.14  124327.718676
hyundai             5371.79  106718.426501
mazda               4059.06  124871.447903
mercedes_benz      30025.54  130852.504838
mini               10541.57   88899.521531
nissan              4669.39  118711.201080
opel                5281.32  129455.924171
peugeot             3065.61  127356.643357
renault             2431.20  128062.365591
seat                4320.17  121768.226333
skoda               6353.54  110993.589744
smart               3518.10  100511.527378
sonstige_autos     44381.44   90467.091295
toyota              5148.00  116219.312602
volkswagen          6600.38  128961.679892
volvo              32914.29  138363.636364


* Mean selling price of the cars sold on ebay are not dependent on odometer. Most of these cars are over 100,000 miles average, mainly between 100,000 - 130000 miles.